ai_graph_color/problem_generator.py

```diff
@@ -83,7 +87,69 @@ def build_graph(points):
     :rtype: graph (adjacency list)
     :return: a planar graph constructed from the given vertices
     """
-    pass
+    lines = create_lines(points)
+    point_distances = create_distance_list(lines, len(points))
+    point_distances = point_distances  # TODO use point_distances
+
+
+def create_lines(points):
+    """
+    Generate a map of all the possible lines between points, indexed by an
+    immutable set of the indexes of the two points connected by the line
+
+    :param points: the points to create lines from
+    :type points: list[tuple(float, float)]
+    :return: the map of lines
+    :rtype: map<frozenset(int), Line>
+    """
+    lines = {}
+    for point_a, point_b in combinations(enumerate(points), 2):
+        connecting_line = Line(point_a[1], point_b[1])
+        key = frozenset([point_a[0], point_b[0]])
```

Comment:
[frozenset](https://docs.python.org/2/library/stdtypes.html#frozenset) is an immutable set